x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/utils/validate.ts

```diff
@@ -51,3 +65,45 @@ export const transformValidateBulkError = (
     });
   }
 };
+
+// Temporary functionality until the new System Actions is in place
+// for now we want to make sure that user cannot configure Isolate action if has no RBAC permissions to do so
+export const validateResponseActionsPermissions = async (
+  securitySolution: SecuritySolutionApiRequestHandlerContext,
+  ruleUpdate: RuleCreateProps | RuleUpdateProps,
+  existingRule?: RuleAlertType | null
+) => {
+  const { experimentalFeatures } = await securitySolution.getConfig();
+
+  if (!experimentalFeatures.endpointResponseActionsEnabled || !isQueryRule(ruleUpdate.type)) {
+    return;
+  }
+
+  const payload = ruleUpdate as QueryRule;
+  const existingPayload = existingRule as Rule<UnifiedQueryRuleParams>;
+
+  if (payload.response_actions?.length || existingPayload?.params?.responseActions?.length) {
+    const endpointAuthz = await securitySolution.getEndpointAuthz();
+
+    const difference = findDifferenceInArrays<ResponseAction, RuleResponseAction>(
+      payload.response_actions,
+      existingPayload?.params?.responseActions
+    );
+
+    difference.forEach((action) => {
+      if ('command' in action?.params) {
+        const isInvalid = !getRbacControl({
+          commandName: getUiCommand(action.params.command),
+          privileges: { ...endpointAuthz, loading: false },
+        });
+
```

Comment:
I would suggest not using `getRbackControl()` since that is working with a UI specific construct as I mentioned earlier.

If you keep this generic as is now, then I would suggest something like:


```typescript
const authzPropName = RESPONSE_CONSOLE_ACTION_COMMANDS_TO_REQUIRED_AUTHZ[
  RESPONSE_ACTION_API_COMMANDS_TO_CONSOLE_COMMAND_MAP[action.params.command]
  ];

const isInvalid = endpiontAuthz[authzPropName];
```

You could also create a `const` that includes the mapping of `RESPONSE_ACTION_API_COMMANDS_TO_REQUIRED_AUTHZ` if that is clearer to understand.


(FYI: @ashokaditya and I have talked about refactoring all of const (there are many for actions) into a nested object that can be use by both the server and UI)

Reply:
👍 Great hint, thank you!